build-tools/automation/build.groovy

```diff
@@ -49,6 +51,22 @@ def publishPackages(filePaths) {
     return status
 }
 
+prInfos = [:]  // Globally defined "static" variable accessible within the hasPrLabel function
+
+def hasPrLabel (gitRepo, prId, prLabel) {
+    if (!prInfos.containsKey(prLabel)) {
+        def curlCommand = "curl https://api.github.com/repos/${gitRepo}/issues/${prId}"
+
+        def grepResult = sh(script: """${curlCommand} | grep '"name": "${prLabel}"' >/dev/null 2>&1""",
+                            returnStatus: true)
+
+        def prInfoContainsLabel = (grepResult == 0)
+        prInfos.put(prLabel, prInfoContainsLabel)
```

Comment:
If we're going to create a map for the labels on the PR, we should make this more efficient in terms of network communication (and GitHub-related `curl` limits!); GitHub can "throttle" these `curl` commands (because we're not using the API key or something), and the previous shell-based effort executed multiple `curl` commands because it was easier/we were lazy.

We have no such reason now; Groovy can consume JSON, and the `curlCommand` URL returns JSON: http://groovy-lang.org/json.html

There's even a [`JsonSlurper.parse(URL)` method](http://docs.groovy-lang.org/latest/html/gapi/groovy/json/JsonSlurper.html#parse(java.net.URL)), so we could avoid the `curl` invocation entirely.

Reply:
Great suggestion! I reduced to one network call to obtain & cache all labels. Had to modify the implementation to use `parseText` instead of parse(url) with a note explaining why. Addressed with commit https://github.com/xamarin/xamarin-android/pull/2767/commits/df2d8be01894b4f157a3fac509f2030f5fe25885